# Translational Bioinformatics in R
Date: Aug 10-14, 2020  
Five-day workshop, primarily in R, occasionally in Python  
Morning (9-12pm): tutorial (required) (Thursday starts at 8:00am)  
Afternoon (1pm - 3pm): lab & QA (optional)

## Table of Contents

* [Instructors & TAs](#instructors--tas)
* [Sessions](#sessions)
    * [Introduction to Big Data](#introduction-to-big-data)
    * [Data manipulation and visualization](#data-manipulation-and-visualization)
    * [Basic statistical analysis](#basic-statistical-analysis)
    * [Machine learning](#machine-learning)
    * [RNA-Seq](#rna-seq)
    * [Single cell RNA-Seq](#single-cell-rna-seq)
    * [From big data to drug discovery](#from-big-data-to-drug-discovery)
    * [Structure-based drug discovery](#structure-based-drug-discovery)
    * [R markdown, R packages, R Shiny](#r-markdown-r-packages-r-shiny)
    * [Q&A, feedback](#qa-feedback)

## Instructors & TAs
+ Bin Chen
+ Ke Liu
+ Jing Xing
+ Eugene Chekalin
+ Mengying Sun
+ Jiayu Zhou (invited)
+ Yuehua Cui (invited)
+ Paul Egeler (invited)
+ Eric Kort (invited)

## Sessions

### [Introduction to Big Data](topic1_big_data_R)

(two sessions, Bin), Day 1: 9-11am

1. Data structure
    + Matrix (data.frame, matrix)
    + Network (igraph, cytoscape, stringdb, KEGG)
    + Unstructured data (text, clinical note, pattern regular expression)
2. Data modality
    + Omics data (genomics, transcriptomics, proteomics, metabolomics)
    + Screening data (CRISPR, pharmacogenomics)
    + Image data (Morphology image)
    + Knowledge graph (PPI)
    + EMR
3. Models
    + cells/organoids/animal models/patients
4. Public data resources
5. Basic R (read/write, data types)
6. Workshop Structure

### [Data manipulation and visualization](topic2_manipulaiton_visualization)

(two sessions, Eugene), Day 1: 11-12pm, Day 2: 9-10am

1. Apply (lapply, sapply, by)
2. Collapse/Summarize data (dplyr)
3. PCA/TSNE/umap (Variation)
4. Clustering (hclust)
5. ggplot2 (boxplot, violin plot, scatter plot, error bar)
6. Heatmap (annotation, clustering, ComplexHeatmap )
7. Publication-ready figures (font, background, legend)

### [Basic statistical analysis](topic3_statistics)

(two sessions, Yuehua) Day 2: 10-12pm

1. Correlation analysis
2. Continuous data (spearman, pearson)
3. Categorical data (fisher test, chi square)
4. Linear regression and logistic regression (odds ratio)
5. Confounding factors
6. P value and FDR (p value correlation)
7. Survival analysis (KM plot, Hazard Ratio)

### [Machine learning](topic4_machine_learning)

(two sessions, Jiayu) Day 3: 9-11am

1. Intro to ML
2. Perceptron and Deep Learning
3. Tree Methods
4. Random Forest
5. Adaboost
6. Gradient Boosting

### [RNA-Seq](topic5_RNASeq)

(two sessions, Ke) Day 3: 11-12, Day 4: 8-9am

1. Sequence alignment
2. DE analysis (edgeR, DESeq, Limma Voom)
3. Enrichment analysis (GSEA, ssGSEA, EnrichR, Hypergeometric test, Pathway Databases)

### [Single cell RNA-Seq](topic6_scRNASeq)

(two sessions, Eric) Day 4: 9-11am

1. Basic (biological theory, common platforms)
2. Sequence analysis (alignment, counting, QC, normalization)
3. Down-stream analysis (dimension reduction, visualization, cell type assignment, and RNA velocity)

### [From big data to drug discovery](topic7_data2drug)

(one session, Bin) Day 4: 11-12

1. Structure representation (SMILES, SD)
2. Structure embedding (fingerprint, pharmacophore)
3. Drug-induced gene expression profiles
4. Systems-based drug discovery (OCTAD)

### [Structure-based drug discovery](topic8_structure_based_drug_design)

(one session, Jing) Day 5: 9-10am

1. Protein structure/3D structure
2. Docking
3. Target fishing

### [R markdown, R packages, R Shiny](topic9_Rmarkdown_Rpackages_Rshiny)

(one session, Paul) Day 5: 10-11am

### Q&A, feedback

(Bin) Day 5: 11:00-12pm
